File: SDKs/amazonaws.com/ec2/2016-11-15/typescript/src/sdk/models/shared/provisionbyoipcidrresult.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// ProvisionByoipCidrResultByoipCidr
/** 
 * Information about the address range.
**/
export class ProvisionByoipCidrResultByoipCidr extends SpeakeasyBase {
  @SpeakeasyMetadata()
  cidr?: Record<string, any>;

  @SpeakeasyMetadata()
  description?: Record<string, any>;

  @SpeakeasyMetadata()
  state?: Record<string, any>;

  @SpeakeasyMetadata()
  statusMessage?: Record<string, any>;
}


export class ProvisionByoipCidrResult extends SpeakeasyBase {
  @SpeakeasyMetadata()
  byoipCidr?: ProvisionByoipCidrResultByoipCidr;
}

File: SDKs/amazonaws.com/ec2/2016-11-15/python/src/sdk/models/shared/provisionbyoipcidrresult.py
import dataclasses
from typing import Any,Optional


@dataclasses.dataclass
class ProvisionByoipCidrResultByoipCidr:
    r"""ProvisionByoipCidrResultByoipCidr
    Information about the address range.
    """
    
    cidr: Optional[dict[str, Any]] = dataclasses.field(default=None)
    description: Optional[dict[str, Any]] = dataclasses.field(default=None)
    state: Optional[dict[str, Any]] = dataclasses.field(default=None)
    status_message: Optional[dict[str, Any]] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class ProvisionByoipCidrResult:
    byoip_cidr: Optional[ProvisionByoipCidrResultByoipCidr] = dataclasses.field(default=None)
    
